inspirehep/records/marshmallow/literature/base.py

```diff
@@ -190,17 +192,22 @@ def get_len_or_missing(maybe_none_list):
         return len(maybe_none_list)
 
     @post_dump
-    def strip_empty(self, data):
-        return strip_empty_values(data)
+    def post_dump(self, data):
+        data = strip_empty_values(data)
+        if is_superuser_or_cataloger_logged_in():
+            data["can_edit"] = True
```

Comment:
why not set can_edit False if it is not superuser or cataloger? Also, why does it need to be in the post dump? Can't it be in the LiteratureMetadataUISchemaV1 as a fields.Method and thus will also show up in the ui_display field and you wouldn't need to duplicate logic in get_ui_display? or am i missing something?

Reply:
because it does not add any useful info :)

Reply:
but you would usually expect to check if a flag is true or false not if it is there or not. It doesn't change anything, I agree. But for me, it makes more sense